### DISCUSSION TITLE
Error "The call is ambiguous between the following methods or properties" Unity3D

### DISCUSSION THREAD

**Discussion by spiNkerZ** [Cysharp/UniTask#470] [Q&A]
Hello everyone Faced with such a problem after upgrading to Unity3D 2023.1.0.b18:

error CS0121: The call is ambiguous between the following methods or properties: 'Cysharp.Threading.Tasks.UnityAsyncExtensions.GetAwaiter(UnityEngine.AsyncOperation)' and 'UnityEngine.AsyncOperationAwaitableExtensions.GetAwaiter(UnityEngine.AsyncOperation)'

My code:

` await UnityEngine.SceneManagement.SceneManager.LoadSceneAsync(ID);`
``
I solved this problem in this way if anyone needs:

```
 AsyncOperation loadSceneHandler = UnityEngine.SceneManagement.SceneManager.LoadSceneAsync(ID);

 await loadSceneHandler.ToUniTask();
```

**Comment by anonymous2585** [comment]
>[!WARNING]
> From the UniTask Readme:
>> In UniTask, await directly uses native timing, while WithCancellation and ToUniTask use specified timing. This is usually not a particular problem, but with LoadSceneAsync, it causes a different order of Start and continuation after await. So it is recommended not to use LoadSceneAsync.ToUniTask.

At first I used this and it seemed to work fine in editor and standalone windows builds. But when I tested in a UWP build (IL2CPP), the `await loadSceneHandler.ToUniTask();` hold the whole app and all I got was a black screen.

**Comment by neuecc** [comment]
UnityAsyncExtensions.GetAwaiter is disabled in `UNITY_2023_1_OR_NEWER`.
https://github.com/Cysharp/UniTask/blob/0c8057c66852e7d720966ef076d4924ebb27e80c/src/UniTask/Assets/Plugins/UniTask/Runtime/UnityAsyncExtensions.cs#L18-L25
Why is it enabled?

**Reply by anonymous2585** [reply]
I don't know...
I'm using Unity 2023.1.16f1 and have the name conflict.
It seems `UNITY_2023_1_OR_NEWER` is false in my project (so the code is included), this is very strange.

**Reply by anonymous2585** [reply]
My issue is with Visual Studio.
The define are not properly set, so Visual Studio underline `await SceneManager.UnloadSceneAsync("my_scene_name");` in red with error CS0121, but it's ok in Unity.

**Reply by neuecc** [reply]
How about regenerating csproj?
Delete the sln and csproj and have Unity regenerate them.

**Reply by anonymous2585** [reply]
Already done, and it doesn't fix preprocessor directives in Visual Studio.
I think it's because of some incompatibility of my Visual Studio version and the Unity Visual Studio package version, but I can't change easily them becuse my company computer doesn't have an access to the internet.
It's not a big issue for me, now I understood it's broken only in Visual Studio.

**Reply by albermotion** [reply]
I know I'm late to the conversation but in case it happens to someone else:
You need to set `using UnityEngine;` in the class you are setting your `await` or else it will try to get the `AsyncOperation` from `LoadSceneAsync` instead of the `Awaitable` version from Unity
